Zoe API Revisions (#773)
Agoric/documentation: main/reference/zoe-api/README.md

Start creating your own contract or build on any of our existing contracts.
Explore our pre-built contracts [here](/guides/zoe/contracts/README.md).

The Zoe API supports the following objects:

| Object | Description |
| --- | --- |
| [Zoe Service](./zoe.md) | Deploy and work with smart contracts. |
| [UserSeat](./user-seat.md) | Used outside contracts to access or manipulate offers. |
| [Zoe Contract Facet](./zoe-contract-facet.md) | Accesses a running contract instance. |
| [ZCFSeat](./zcfseat.md) | Used within contracts to access or manipulate offers. |
| [ZCFMint](./zcfmint.md) | Used by the **[Zoe Contract Facet](./zoe-contract-facet.md)** to issue digital assets. |

> | [Price Authority](./price-authority.md) | Gives reliable quotes for prices. |

Author: Fixed.

> The Zoe API uses the following libraries:

Author: Fixed.

| Library | Description |
| --- | --- |
| [ZoeHelpers](./zoe-helpers.md) | Functions that extract common contract code and patterns into reusable helpers. |
| [Ratio Math](./ratio-math.md) | Functions that let you create and manipulate **[Ratios](./zoe-data-types.md#ratio)**. |

The Zoe API introduces and uses the following data types:

| Data Type | Description |
| --- | --- |

> | [Allocation](./zoe-data-types.md#allocation) | The **[Amounts](/reference/ertp-api/ertp-data-types.md#amount)** to be paid out to each seat upon exiting a **Proposal** |

Author: Fixed.

> | [AmountKeywordRecord](./zoe-data-types.md#amountkeywordrecord) | Records in which the keys are keywords and the values are **[Amounts](/reference/ertp-api/ertp-data-types.md#amount)**. |

Author: Very good point; keywords need clear defining. I've added "Keyword" to the "Zoe Data Types" topic, as well as making the suggested changes to the README.md file.

As for the explanation as to why we require them to start with an upper case letter, that seems out of scope for technical docs. It's interesting info, but it seems like unnecessary clutter for most (if not all) readers.

> | [AmountKeywordRecord](./zoe-data-types.md#amountkeywordrecord) | Records in which the keys are keywords and the values are **[Amounts](/reference/ertp-api/ertp-data-types.md#amount)**. |

Reviewer: cf. https://docs.agoric.com/glossary/#keyword

> | [AmountKeywordRecord](./zoe-data-types.md#amountkeywordrecord) | Records in which the keys are keywords and the values are **[Amounts](/reference/ertp-api/ertp-data-types.md#amount)**. |

Author: Yeah, thanks for pointing that out. At some point I'm going to have to rationalize & revise the glossary, but for now I'm pretty much ignoring its existence. (I suspect the glossary will be deleted, with term definitions instead being given within the body of relevant texts or in data type topics like the ones that now appear in the ERTP API & Zoe API sections.)

> | [AmountKeywordRecord](./zoe-data-types.md#amountkeywordrecord) | Records in which the keys are keywords and the values are **[Amounts](/reference/ertp-api/ertp-data-types.md#amount)**. |

Reviewer: > I suspect the glossary will be deleted, with term definitions instead being given within the body of relevant texts or in data type topics like the ones that now appear in the ERTP API & Zoe API sections

I hope not! The glossary is a good (and in fact the current best) entry point for finding those relevant detail pages.

> | [Instance](./zoe-data-types.md#instance) | Opaque objects that represent contract instances. |

Reviewer 2: ```suggestion
| [Instance](./zoe-data-types.md#instance) | A **Handle** (opaque object) that represents a contract instance. |
```

Author: Good clarification. Text updated.

> | [InvitationIssuer](./zoe-data-types.md#invitationissuer) | Special types of **[Issuers](/reference/ertp-api/issuer.md)** that represent the **Issuer** that has the power to generate **Invitations** for the contract. |

Reviewer 2: ```suggestion
| [InvitationIssuer](./zoe-data-types.md#invitationissuer) | An **[Issuer](/reference/ertp-api/issuer.md)** for **Invitations** -- the right to participate in a contract. |
```

Author: I changed the description of InvitationIssuer to:  " An Issuer for Invitations, which grant the right to participate in a contract. "
